docssrc/src/commandarguments.md

```diff
@@ -0,0 +1,202 @@
+- [Access unsafe arguments](#access-unsafe-arguments)
+  - [Access arguments by node name](#access-arguments-by-node-name-1)
+  - [Access arguments by index](#access-arguments-by-index-1)
+
+ATTENTION!!!!! Every executable and generated documentation has to be deleted before committing anything!
+
+-----
+
+## Access the inner structure directly
+
+To provide arguments, the `CommandArguments` class stores:
+
+- a `Object[]` of parsed arguments
+- a `Map<String, Object>` of parsed arguments mapped to their node names
+- a `String[]` of raw arguments
+- a `Map<String, String>` of raw arguments mapped to their node names
+- a `String` which holds the full input
+
+Although not recommended, it is possible to access these fields directly with methods the `CommandArguments` class provides:
+
+```java
+Object[] args();                  // Returns the argument array
+Map<String, Object> argsMap();    // Returns an unmodifiable map containing the arguments mapped to their node names
+String[] rawArgs();               // Returns the raw argument array
+Map<String, String> rawArgsMap(); // Returns an unmodifiable map containing the raw arguments mapped to their node names
+String getFullInput();               // Returns the full command string (including the /)
+```
+
+Additionally, the `CommandArguments` class has one more method that isn't directly backed by a field and returns the amount of arguments for a command:
+
+```java
+int count();
+```
+
+While these methods can be used to access arguments, it may be safer to use the other methods the `CommandArguments` class provides to access arguments.
+
+-----
+
+## What terms are used?
+
+Throughout this page, multiple terms are used that may need an explanation.
+
+### `nodeName`
+
+The `nodeName` is set when initializing an argument. For example:
+
+```java
+new StringArgument("string") 
+```
+
+The `nodeName` here would be `string`.
+
+### `raw argument`
+
+A "raw argument" is the `String` form of an argument as written in a command. For example:
+
+A user defines a command `/mycommand` that accepts a `double` as the first argument and an entity selector as the second argument. It could be executed with the values `15.3` as the `double` value and `@e` as the entity selector:
+
+```mccmd
+/mycommand 15.3 @e
+```
+
+When [accessing the raw arguments](#access-raw-arguments) of this command there are `15.3` and `@e` available as `String`s.
+
+However, when [accessing the arguments](#access-arguments) of this command there is `15.3` available as `double` and `@e` available as `Collection<Entity>`.
+
+### `unsafe argument`
+
+When [accessing arguments](#access-arguments) you need to cast the `Object` returned by these methods to the type the argument returns. More about casting arguments [here](./arguments.md#argument-casting).
+
+However, the `CommandArguments` class provides a way to remove the need to cast the arguments which is referred to as `unsafe arguments` in this page.
```

Comment:
Unsafe arguments should have an example (similar to how the raw arguments has an example) showcasing how it differs to accessing arguments normally. It should also point out the issues with it (e.g. doesn't work with Kotlin's inference system).